compiler/rustc_incremental/src/persist/work_product.rs

```diff
@@ -46,7 +46,12 @@ pub fn copy_cgu_workproduct_to_incr_comp_cache_dir(
 
 /// Removes files for a given work product.
 pub fn delete_workproduct_files(sess: &Session, work_product: &WorkProduct) {
-    for (_, path) in &work_product.saved_files {
+    for path in work_product
+        .saved_files
+        .items()
+        .map(|(_, path)| path.as_str())
+        .into_sorted_stable_ord(false)
```

Comment:
`&String` should implement `StableOrd`

Reply:
Hmm, apparently not. Would it be correct to impl StableOrd for `&T` if `T` is StableOrd? I believe it should but just checking in (if not, then I'll just add the impl for `&String` on its own)

Reply:
> Would it be correct to impl StableOrd for &T if T is StableOrd?

yes